socorro/unittest/lib/test_datetimeutil.py

```diff
@@ -27,172 +30,155 @@ def test_string_to_datetime():
 
 
 def test_date_to_string():
     # Datetime with timezone
     date = datetime.datetime(2012, 1, 3, 12, 23, 34, tzinfo=UTC)
     res_exp = '2012-01-03T12:23:34+00:00'
     res = datetimeutil.date_to_string(date)
-    eq_(res, res_exp)
+    assert res == res_exp
 
     # Datetime without timezone
     date = datetime.datetime(2012, 1, 3, 12, 23, 34)
     res_exp = '2012-01-03T12:23:34'
     res = datetimeutil.date_to_string(date)
-    eq_(res, res_exp)
+    assert res == res_exp
 
     # Date (no time, no timezone)
     date = datetime.date(2012, 1, 3)
     res_exp = '2012-01-03'
     res = datetimeutil.date_to_string(date)
-    eq_(res, res_exp)
+    assert res == res_exp
 
 
-@raises(TypeError)
 def test_date_to_string_fail():
-    datetimeutil.date_to_string('2012-01-03')
+    with pytest.raises(TypeError):
+        datetimeutil.date_to_string('2012-01-03')
 
 
 def test_uuid_to_date():
     uuid = "e8820616-1462-49b6-9784-e99a32120201"
     date_exp = datetime.date(year=2012, month=2, day=1)
     date = datetimeutil.uuid_to_date(uuid)
-    eq_(date, date_exp)
+    assert date == date_exp
 
     uuid = "e8820616-1462-49b6-9784-e99a32181223"
     date_exp = datetime.date(year=1118, month=12, day=23)
     date = datetimeutil.uuid_to_date(uuid, century=11)
-    eq_(date, date_exp)
+    assert date == date_exp
 
 
 def test_date_to_weekly_partition_with_string():
     datestring = '2015-01-09'
     partition_exp = '20150105'
 
     partition = datetimeutil.datestring_to_weekly_partition(datestring)
-    eq_(partition, partition_exp)
+    assert partition == partition_exp
 
     # Is there a better way of testing that we handle 'now' as a date value?
     datestring = 'now'
     date_now = datetime.datetime.now().date()
     partition_exp = (date_now + datetime.timedelta(0 - date_now.weekday())).strftime('%Y%m%d')
     partition = datetimeutil.datestring_to_weekly_partition(datestring)
-    eq_(partition, partition_exp)
-
-
-def test_date_to_weekly_partition_with_datetime():
-    proposed_and_expected = (
-        (datetime.datetime(2014, 12, 29), '20141229'),
-        (datetime.datetime(2014, 12, 30), '20141229'),
-        (datetime.datetime(2014, 12, 31), '20141229'),
-        (datetime.datetime(2015,  1,  1), '20141229'),
-        (datetime.datetime(2015,  1,  2), '20141229'),
-        (datetime.datetime(2015,  1,  3), '20141229'),
-        (datetime.datetime(2015,  1,  4), '20141229'),
-        (datetime.datetime(2015,  1,  5), '20150105'),
-        (datetime.datetime(2015,  1,  6), '20150105'),
-        (datetime.datetime(2015,  1,  7), '20150105'),
-        (datetime.datetime(2015,  1,  8), '20150105'),
-        (datetime.datetime(2015,  1,  9), '20150105'),
-    )
-    for to_be_tested, expected in proposed_and_expected:
-        result = datetimeutil.datestring_to_weekly_partition(to_be_tested)
-        eq_(result, expected)
+    assert partition == partition_exp
+
+
+@pytest.mark.parametrize('data, expected', [
```

Comment:
huh. I had to look up parametrize and it is absolutely the right thing here

Reply:
parametrize is super awesome because each individual iteration becomes a separate test case which can be executed specifically.

Having said that, it's sometimes tricky to figure out what the id is based on the data.

So I wrote pytest-wholenodeid which helps. https://pypi.python.org/pypi/pytest-wholenodeid